lib/charms/opensearch/v0/opensearch_plugins.py

```diff
@@ -278,32 +294,42 @@ def _on_update_status(self, event):
 class OpenSearchPluginError(OpenSearchError):
     """Exception thrown when an opensearch plugin is invalid."""
 
-
-class OpenSearchPluginRelationClusterNotReadyError(OpenSearchPluginError):
-    """Exception thrown when making API calls and cluster is not yet ready.
-
-    This exception in specific should be handled by classes managing the relations of plugins.
-    """
+    def __init__(self, msg):
+        super().__init__(msg)
 
 
 class OpenSearchPluginMissingDepsError(OpenSearchPluginError):
     """Exception thrown when an opensearch plugin misses installed dependencies."""
 
+    def __init__(self, name, deps):
+        super().__init__(f"Failed to install plugin: {name} - missing dependencies {deps}")
+        self.deps = deps
```

Comment:
safe to delete - unless you see a use for them besides logging?

Reply:
Keeping only the fields, so we not forget them

Reply:
It seems that we don't need to keep them. They're only used for logging. Let's remove this constructor altogether and just set the right message with the list of dependencies in the str.